packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContent/ConditionBuilderContent.js

```diff
@@ -31,53 +31,38 @@ const ConditionBuilderContent = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [rootState, conditionBuilderRef]);
 
-  useEffect(() => {
-    if (isConditionBuilderActive == false) {
-      if (conditionBuilderRef.current) {
-        const initial = conditionBuilderRef.current.querySelector(
-          '[role="gridcell"] button'
-        );
-
-        if (initial) {
-          initial.setAttribute('tabindex', '0');
-        }
-      }
-    }
-  }, [isConditionBuilderActive, conditionBuilderRef]);
   const onStartConditionBuilder = () => {
     //when add condition button is clicked.
-    setIsConditionBuilderActive(false);
+    setIsConditionBuilderActive(true);
     setRootState(initialState ?? emptyState); //here we can set an empty skeleton object for an empty condition builder,
     //or we can even pre-populate some existing builder and continue editing
   };
 
   const onRemove = useCallback(
-    (groupIndex) => {
+    (groupId) => {
       setRootState({
         ...rootState,
-        groups: rootState.groups.filter(
-          (group, gIndex) => groupIndex !== gIndex
-        ),
+        groups: rootState.groups.filter((group) => groupId !== group.id),
       });
     },
     [setRootState, rootState]
   );
 
-  const onChangeHandler = (updatedGroup, groupIndex) => {
+  const onChangeHandler = (updatedGroup, groupId) => {
     /**
      * This method is triggered from inner components. This will be called every time when any change is to be updated in the rootState.
      * This gets the updated group as argument.
      */
     setRootState({
       ...rootState,
-      groups: rootState.groups.map((group, gIndex) =>
-        groupIndex === gIndex ? updatedGroup : group
+      groups: rootState.groups.map((group) =>
```

Comment:
since we're only updating one object per change we don't need to use map here to iterate over every item in the array. it would be more efficient to get the index of the group in the array. consider this code
```
    const groupIdx = rootState.groups.findIndex((o) => o.id === groupId);
    const groups = [
      ...rootState.groups.slice(0, groupIdx),
      updatedGroup,
      ...rootState.groups.slice(groupIdx + 1),
    ];
    setRootState({
      ...rootState,
      groups,
    });
```
with code like this we find the index and replace it directly without the need to map over everything else.

Reply:
modified by using slice().